src/cli/CLIParsing.h

```diff
@@ -57,12 +62,31 @@ static AppType runType(QCommandLineParser& parser) {
     // Search for the extension ID among the start-up arguments. This is the only thing
     // the native messaging APIs of Firefox and Chrome have in common. See also:
     // https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Native_messaging#extension_side
+
+    Settings settings;
+
+    QRegularExpression re("^chrome-extension://(.+?)/$");
+
     for (auto&& path : parser.positionalArguments()) {
-        if (kNativeMessagingClients.contains(path)) {
+        // Matches anything Firefox based
+        if (settings.nativeMessagingClients().contains(path)) {
+            return NativeMsg;
+        }
+
+        // Matching anything Chromium based
+        auto match = re.match(path);
+        if (match.hasMatch() && settings.nativeMessagingClients().contains(match.captured(1))) {
             return NativeMsg;
         }
     }
 
+    // TODO: if this program is started as a browser's native messaging client,
```

Comment:
Have you tested this, or are we leaving it as a TODO

Reply:
Good catch. I haven't tried it, will do so before I merge.